packages/react-router-dom/dom.ts

```diff
@@ -109,6 +109,16 @@ export function getSearchParamsForLocation(
   return searchParams;
 }
 
+export type SubmitTarget =
+  | HTMLFormElement
+  | HTMLButtonElement
+  | HTMLInputElement
+  | FormData
+  | URLSearchParams
+  | { [name: string]: string }
+  | NonNullable<unknown> // Raw payload submissions
```

Comment:
We can now submit anything other than `undefined`

Reply:
@ryanflorence - Mark brought up a good point that this could potentially open up a footgun for users thinking anything can be serialized into `FormData`.  With `encType:null` we can accept _anything_ (and rule out undefined for clarity).

But without `encType` not everything _can_ be encoded into `FormData`

```
// Technically arrays are fine and serialize into formData as a map of index => value
submit([1,2,3])

// but strings will blow up without any type hints
submit("plain text")

// as will nested objects and other complex structures
submit({ parent: { child: { key: "value" } } })
```

Right now the plan is to handle any advanced type inference in a new PR so we can tackle it there based on what we decide.  We could also just log a warning if you submit a non-FormData serializable value?